test/000_fullchain.test.ts

```diff
@@ -174,44 +173,380 @@ describe('Integration tests', function () {
             //   - Worker:
             //      - balance: balance before + taskStake + taskReward
             //      - frozen: frozen before - taskStake
-            await checkBalancesAndFrozens({
-                proxyBalance:
-                    dealPrice +
-                    schedulerStakePerDeal -
-                    (taskPrice + schedulerStakePerTask) * completedTasks,
-                accounts: [
-                    { signer: sponsor, balance: 0, frozen: dealPrice - taskPrice * completedTasks },
-                    { signer: requester, balance: 0, frozen: 0 },
-                    {
-                        signer: scheduler,
-                        balance: (schedulerStakePerTask + schedulerRewardPerTask) * completedTasks,
-                        frozen: schedulerStakePerDeal - schedulerStakePerTask * completedTasks,
-                    },
-                    { signer: appProvider, balance: appPrice * completedTasks, frozen: 0 },
-                    { signer: datasetProvider, balance: datasetPrice * completedTasks, frozen: 0 },
-                    {
-                        signer: worker1,
-                        balance: (workerStakePerTask + workerRewardPerTask) * completedTasks,
-                        frozen: 0,
-                    },
+            // Verify token balance changes
+            const expectedProxyBalanceChange = -(
+                taskPrice * completedTasks +
+                schedulerStakePerTask * completedTasks
+            );
+            expect(finalizeTx).to.changeTokenBalances(
+                iexecPoco,
+                [proxyAddress, sponsor, requester, scheduler, appProvider, datasetProvider],
+                [
+                    expectedProxyBalanceChange, // Proxy
+                    -taskPrice * completedTasks, // Sponsor
+                    0, // Requester
+                    schedulerStakePerTask + schedulerRewardPerTask, // Scheduler
+                    appPrice, // AppProvider
+                    datasetPrice, // DatasetProvider
                 ],
+            );
+            for (const worker of workers) {
+                expect(finalizeTx).to.changeTokenBalances(
+                    iexecPoco,
+                    [worker],
+                    [workerStakePerTask + workerRewardPerTask / workers.length],
+                );
+            }
+            // Calculate expected frozen changes
+            const expectedFrozenChanges = [
+                0, // Proxy
+                -taskPrice * completedTasks, // Sponsor
+                0, // Requester
+                -schedulerStakePerTask * completedTasks, // Scheduler
+                0, // AppProvider
+                0, // DatasetProvider
+            ];
```

Comment:
```suggestion
                0, // DatasetProvider
                ...workers.map(() => 0) // Add 0 for each worker
            ];
```
And remove the for loop below.

Reply:
[push workers frozen balance in the array directly](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/387e4a4ccf72da1c1be89d9bad674f35ce8bd139)